pkg/model/iam/iam_builder.go

```diff
@@ -329,9 +583,207 @@ func (b *IAMPolicyResource) Open() (io.Reader, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error building IAM policy: %v", err)
 	}
-	json, err := policy.AsJSON()
+	j, err := policy.AsJSON()
 	if err != nil {
 		return nil, fmt.Errorf("error building IAM policy: %v", err)
 	}
-	return bytes.NewReader([]byte(json)), nil
+	return bytes.NewReader([]byte(j)), nil
+}
+
+func addECRPermissions(p *IAMPolicy) {
+	// TODO - I think we can just have GetAuthorizationToken here, as we are not
+	// TODO - making any API calls except for GetAuthorizationToken.
+
+	// We provide ECR access on the nodes (naturally), but we also provide access on the master.
+	// We shouldn't be running lots of pods on the master, but it is perfectly reasonable to run
+	// a private logging pod or similar.
+	// At this point we allow all regions with ECR, since ECR is region specific.
+	p.Statement = append(p.Statement, &IAMStatement{
```

Comment:
Moving these around makes them tricky to review.  But if we don't end up having the duplicate function, we might not need to move them (or we can move them in a set of small PRs)